cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,210 @@
+}
+
+void SpoilerBackgroundUpdater::downloadFromURL(QUrl url, bool saveResults)
+{
+    auto *nam = new QNetworkAccessManager(this);
+    QNetworkReply *reply = nam->get(QNetworkRequest(url));
+
+    if (saveResults)
+    {
+        // This will write out to the file (used for spoiler.xml)
+        connect(reply, SIGNAL(finished()), this, SLOT(actDownloadFinishedSpoilersFile()));
+    }
+    else
+    {
+        // This will check the status (used to see if we're in spoiler season or not)
+        connect(reply, SIGNAL(finished()), this, SLOT(actCheckIfSpoilerSeasonEnabled()));
+    }
+}
+
+void SpoilerBackgroundUpdater::actDownloadFinishedSpoilersFile()
+{
+    // Check for server reply
+    auto *reply = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = reply->error();
+
+    if (errorCode == QNetworkReply::NoError)
+    {
+        spoilerData = reply->readAll();
+        reply->deleteLater();
+
+        // Save the spoiler.xml file to the disk
+        saveDownloadedFile(spoilerData);
+    }
+    else
+    {
+        qDebug() << "Error downloading spoilers file" << errorCode;
+    }
+}
+
+void SpoilerBackgroundUpdater::actCheckIfSpoilerSeasonEnabled()
+{
+    auto *response = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = response->error();
+
+    if (errorCode == QNetworkReply::ContentNotFoundError)
+    {
+        // Spoiler season is offline at this point, so the spoiler.xml file can be safely deleted
+        // The user should run Oracle to get the latest card information
+        QString fileName = settingsCache->getSpoilerCardDatabasePath();
+        QFileInfo fi(fileName);
+        QDir fileDir(fi.path());
+        QFile file(fileName);
+
+        // Delete the spoiler.xml file as we're not in spoiler season
+        if (file.exists() && file.remove())
+        {
+            qDebug() << "Spoiler Season Offline, Deleting spoiler.xml";
+        }
+
+        /*
+         * ALERT: Ensure two reloads of the card database do not happen
+         * at the same time or a racetime condition can/will happen!
+         */
+        qDebug() << "Spoiler Season Offline, Reloading Database";
+        QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+    }
+    else if (errorCode == QNetworkReply::NoError)
+    {
+        qDebug() << "Spoiler Service Online";
+        startSpoilerDownloadProcess(SPOILERS_URL, true);
+    }
+    else
+    {
+        qDebug() << "Error: Spoiler download failed with reason" << errorCode;
+    }
+}
+
+bool SpoilerBackgroundUpdater::saveDownloadedFile(QByteArray data)
+{
+    QString fileName = settingsCache->getSpoilerCardDatabasePath();
+    QFileInfo fi(fileName);
+    QDir fileDir(fi.path());
+
+    if (!fileDir.exists() && !fileDir.mkpath(fileDir.absolutePath()))
+    {
+        return false;
+    }
+
+    // Check if the data matches. If it does, then spoilers are up to date.
+    if (getHash(fileName) == getHash(data))
```

Comment:
Are you hashing the file name or the file contents? Maybe include the individual hash values in a qDebug?

Reply:
nvm I see it's overloaded. That works